test/integration/scheduler/preemption_test.go

```diff
@@ -793,13 +827,31 @@ func TestPDBInPreemption(t *testing.T) {
 				t.Fatalf("Error creating node %v: %v", nodeConf.name, err)
 			}
 		}
+
+		pods := make([]*v1.Pod, len(test.existingPods))
+		var err error
+		// Create and run existingPods.
+		for i, p := range test.existingPods {
+			if pods[i], err = runPausePod(cs, p); err != nil {
+				t.Fatalf("Test [%v]: Error running pause pod: %v", test.description, err)
+			}
+			addPodConditionReady(p)
+			if _, err := context.clientSet.CoreV1().Pods(context.ns.Name).UpdateStatus(p); err != nil {
+				t.Fatal(err)
+			}
+		}
+
 		// Create PDBs.
 		for _, pdb := range test.pdbs {
 			_, err := context.clientSet.PolicyV1beta1().PodDisruptionBudgets(context.ns.Name).Create(pdb)
 			if err != nil {
 				t.Fatalf("Failed to create PDB: %v", err)
 			}
 		}
+		// Wait for PDBs to be stable
+		for i, pdb := range test.pdbs {
+			waitPDBStable(t, context.clientSet, test.pdbPodNum[i], context.ns.Name, pdb.Name)
```

Comment:
Should we check pdb stable in scheduler cache directly?

Reply:
Good point. I think that is a safer approach to ensure that PDB is updated when scheduler processes it. I will make the change.